packages/data/README.md

```diff
@@ -1028,6 +1028,52 @@ function Component() {
 }
 ```
 
+### Normalizing Selector Arguments
+
+In specific circumstances it may be necessary to normalize the arguments passed to a given call of a selector. This is necessary because selectors are pre-bound to resolvers before they applied. Moreover, metadata selectors/resolvers are also created using the supplied arguments.
+
+In these circumstances selectors may optionally define a custom property ("method") called `normalizeArgs` which should be a function thats returns the normalized form of the arguments.
+
+For example, if the 3rd argument of the selector is supposed to be an `Number` but it is called with a `String` then normalizeArgs can be defined to coerce the argument type:
+
+```js
+
+const getItemsSelector = ( name, type, id ) => {
+	// here 'id' is now guaranteed to be a number.
+	return state.items[name][type][id] || null;
+}
+
+const getItemsResolver = ( name, type, id ) => {
+	// 'id' is also guaranteed to be a number in the resolver.
+	return {};
+}
+
+// Define normalization function.
+getItemsSelector.normalizeArgs = ( args ) {
```

Comment:
I'm not sure I enjoy the inconsistency here. Perhaps if we spread the `args` when `normalizeArgs` is invoked then we could destructure them which might make things clearer.

```js
getItemsSelector.normalizeArgs = ( { name, type, id } ) {
//
```